### PR TITLE
log packing ratio progress

### PR DESCRIPTION
**Pull request by milocress**
Logs packing ratio

log-test-ZUxVSC

### CODE REVIEW

File: llmfoundry/data/packing.py [hunk 1]
```diff
@@ -8,6 +8,8 @@
 import numpy as np
 import torch
 from omegaconf import DictConfig
+from tqdm import tqdm
+from tqdm.contrib.logging import logging_redirect_tqdm
```
Comment (dakinggg) [inline llmfoundry/data/packing.py:12]: woah what is this !?

Comment (milocress) [reply llmfoundry/data/packing.py:12]: progress bar piped to logs!

**Comment by milocress** [comment]
> Could you share a run so I can see the logs produced by this?

log-test-DOHCxP

File: llmfoundry/data/packing.py [hunk 2]
```diff
@@ -350,6 +354,7 @@ def auto_packing_ratio(dataloader_cfg: DictConfig,
     # Restore rng state.
     reproducibility.load_rng_state(rng_state)
 
+    log.debug(f'Found packing ratio: {packing_ratio}')
```
Comment (irenedea) [inline llmfoundry/data/packing.py:357]: We're already logging this in dataloader.py. I think you can remove this one. Would be nice to make the log in dataloader.py a `log.debug`

File: llmfoundry/data/packing.py [hunk 3]
```diff
@@ -447,6 +451,10 @@ def profile(raw_batch_size: int) -> Tuple[Optional[float], Optional[float]]:
         waste_percent = 100 * packer.waste
         return padding_percent, waste_percent
 
-    for packing_ratio, raw_batch_size in zip(packing_ratios, raw_batch_sizes):
-        padding, waste = profile(raw_batch_size)
-        yield (packing_ratio, padding, waste)
+    with logging_redirect_tqdm(loggers=[log]):
```
Comment (dakinggg) [inline llmfoundry/data/packing.py:454]: Not sure this worked? At least I don't see any new tqdms in the logs for this search

Comment (milocress) [reply llmfoundry/data/packing.py]: ah this one has the tqdm :) `log-test-qKA2k1`

**Comment by dakinggg** [comment]
@milocress I don't think this quite worked? Maybe it would be simpler to avoid trying to get tqdm to work fully here and just print out the packing ratios (or some subset) tried?
```
2024-04-02 21:45:51,223: rank0[1257][MainThread]: INFO: llmfoundry.data.packing: Profiling packing ratios
Profiling packing ratios: 0it [00:00, ?it/s]ead]: DEBUG: llmfoundry.data.packing:
Profiling packing ratio 1.0: 0it [00:00, ?it/s]]: DEBUG: llmfoundry.data.packing:
Profiling packing ratio 1.0: 1it [00:00,  3.96it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 2.0: 1it [00:00,  3.96it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 2.0: 2it [00:00,  4.29it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 3.1: 2it [00:00,  4.29it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 3.1: 3it [00:00,  4.53it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 4.1: 3it [00:00,  4.53it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 4.1: 4it [00:01,  3.55it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 5.1: 4it [00:01,  3.55it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 5.1: 5it [00:01,  4.03it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 6.1: 5it [00:01,  4.03it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 6.1: 6it [00:01,  4.40it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 7.2: 6it [00:01,  4.40it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 7.2: 7it [00:01,  4.70it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 8.2: 7it [00:01,  4.70it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 8.2: 8it [00:01,  4.94it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 9.2: 8it [00:01,  4.94it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 9.2: 9it [00:01,  5.11it/s]EBUG: llmfoundry.data.packing:
Profiling packing ratio 10.2: 9it [00:01,  5.11it/s]BUG: llmfoundry.data.packing:
Profiling packing ratio 10.2: 10it [00:02,  5.23it/s]UG: llmfoundry.data.packing:
Profiling packing ratio 11.3: 10it [00:02,  5.23it/s]UG: llmfoundry.data.packing:
Profiling packing ratio 11.3: 11it [00:02,  5.30it/s]UG: llmfoundry.data.packing:
Profiling packing ratio 12.3: 11it [00:02,  5.30it/s]UG: llmfoundry.data.packing:
2024-04-02 21:45:54,241: rank0[1257][MainThread]: DEBUG: llmfoundry.data.packing:
Profiling packing ratio 12.3: 11it [00:03,  3.65it/s]UG: llmfoundry.data.packing:
2024-04-02 21:45:54,241: rank0[1257][MainThread]: DEBUG: llmfoundry.data.packing:

2024-04-02 21:45:54,266: rank0[1257][MainThread]: INFO: llmfoundry.data.finetuning.dataloader: Using packing ratio 11.3
```

**Comment by milocress** [comment]
> @milocress I don't think this quite worked? Maybe it would be simpler to avoid trying to get tqdm to work fully here and just print out the packing ratios (or some subset) tried?

You're right, may have been overthinking it with the progress bar - I'll have it print the packing ratios and the progress percentage and call it a day

**Comment by milocress** [comment]
@dakinggg here is the updated manual run: `log-test-kvSMKt`

File: llmfoundry/data/packing.py [hunk 4]
```diff
@@ -8,6 +8,7 @@
 import numpy as np
 import torch
 from omegaconf import DictConfig
+from tqdm import tqdm
```
Comment (dakinggg) [inline llmfoundry/data/packing.py:11]: remove

**Review by dakinggg — APPROVED** [review]

**Pull request merged**